### ISSUE TITLE
A1 Web option requirements clarifications

### ISSUE THREAD

**Issue by chai-tang** [CISC-CMPE-327/Information#18]
What's the different between these /login requirements?
R1.3 "If the user has logged in, redirect to the user profile page" 
and 
R1.10 "If email/password are correct, redirect to /"

Are these not both asking that a successful login brings the user to the main profile page? What's the relevant distinction here?

**Comment by steven-hh-ding** [comment]
Thanks!

```
R1.3 "If the user has logged in, redirect to the user profile page"
```
So if the user has already logged in, and he/she requests the /login page, then will redirect him/her to the / profile page. So we don't allow the user to use the /login form again.

```
R1.10 "If email/password are correct, redirect to /"
```
So here the user has not logged in and is able to access the login form, then we go through the password/email validation process. 

It is like if you already logged in facebook, if you go to the facebook login page it will direct you to the home page rather than showing you the login form.

**Comment by chai-tang** [comment]
Thanks for the response! That makes sense.

One more question: For the email/password related requirements, are we meant test these with a series of pre-determined invalid inputs or should it be a more thorough validation than that? Would checking for pattern attributes in the input fields be a good test?

**Comment by steven-hh-ding** [comment]
yes it is supposed to test with valid/invalid/edge cases with specific input and output. 

```
Would checking for pattern attributes in the input fields be a good test?
```
for me this sounds more like how you implement the input validation in the application (the feature), rather than testing.

**Comment by chai-tang** [comment]
Understood, thanks!
